### ISSUE TITLE
Our new colouring support can overwrite "new".

### ISSUE THREAD

**Issue by skx** [lumail/lumail2#62]
At the moment we blindly strip any existing colours when we set the colour for a line.

We should ensure that we never overwrite the `unread` status of a line.

To do that define a colour for unread-status, and use that instead of RED, BLUE, etc.

i.e

       Config:set("global.unread") = "red"

Then for unread things set:

          --
          -- If the message is unread then show it in RED
          --
          if ( string.find( msg:flags(), "N" ) ) then
                    output = "$[UNREAD]" .. output
          end

Finally we ensure that strip_colours is updated to never remove `UNREAD` as colour.

**Comment by skx** [comment]
We should also allow changing colour mid-line, as per the comment(s) on the end of #61.

**Comment by skx** [comment]
OK I've committed [an example program](https://github.com/lumail/experiments/tree/master/colour-string) of how I'm going to handle this, but I'm rushed tonight so I'll do no further work.  

The intention is to allow a user to specify a string like so:

     $[RED]dragons $[GREEN]grass $[YELLOW]sun

We want to split that up into tokens with the correct colour associated with each string.  The following is an example:

      $ ./colour-string '$[RED]dragons $[GREEN]grass $[YELLOW]sun'
      RED string: 'dragons '
      GREEN string: 'grass '
      YELLOW string: 'sun'

This seems to handle the cases that I cared about:

* A string with no formatting - is left alone.
* A string with a leading, single, colour is in that colour for its full length.
* A string with multiple colours is split into multiple chunks.

Input welcome on that sample code.  Otherwise I'll port it over tomorrow.
